### PR TITLE
build: enable support for Kotlin multiplatform subprojects

### PR DESCRIPTION
**Pull request by AngeloFilaseta**
It will be possible to include a multiplatform subproject in the repository using the kotlin("multiplatform") plugin and a "multiplatform" property.

### CODE REVIEW

**Comment by DanySK** [comment]
@AngeloFilaseta for future commits, the simple present is preferable.
Is it a problem for you if I squash this in after review?

**Comment by AngeloFilaseta** [comment]
@DanySK Of course, no problem for the squash. I'll be more careful for commit messages.

**Review by DanySK — CHANGES_REQUESTED** [review]
Two very minor changes / discussions to be solved before merge.

File: buildSrc/src/main/kotlin/Util.kt [hunk 1]
```diff
@@ -119,4 +119,11 @@ object Util {
      * Directly accesses the plugin id.
      */
     val Provider<PluginDependency>.id get() = get().pluginId
+
+    /**
+     * Checks if the project contains a property with key "multiplatform" and if its value is true.
+     * @return true if the property exists and it is true, false otherwhise.
+     */
+    val Project.isMultiplatform get() =
+        this.hasProperty("multiplatform") && this.property("multiplatform") == "true"
```
[the remark this answers is not in the document]
Comment (AngeloFilaseta) [reply buildSrc/src/main/kotlin/Util.kt:128]: Perfect, will do.

File: alchemist-full/build.gradle.kts [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+import Util.isMultiplatform
+
 plugins {
     application
 }
 
 dependencies {
     runtimeOnly(rootProject)
-    rootProject.subprojects.filterNot { it == project }.forEach {
+    rootProject.subprojects.filterNot { it == project || it.isMultiplatform }.forEach {
```
[the remark this answers is not in the document]
Comment (AngeloFilaseta) [reply alchemist-full/build.gradle.kts:18]: I think it is possible to achieve this goal, but not by adding it to the dependencies like this. I'll check for some possible solutions. The jar generated by my project contains the frontend compiled code too, as it serves it to the clients. However, this is a special case. A multiplatform project in the future could build its jar or used the compiled code in a different way. I'll think about it tomorrow.

**Review by DanySK — APPROVED** [review]

**Pull request merged**